test/set_game/game_server_test.exs

```diff
@@ -288,4 +297,62 @@ defmodule SetGame.GameServerTest do
       assert {:set_called, player.id} == GameServer.state(pid)
     end
   end
+
+  describe "#get_uniq_name" do
+    test "returns the uniq name for a via tuple" do
+      assert "hola" == GameServer.get_uniq_name(GameServer.via_tuple("hola"))
+    end
+
+    test "returns the uniq name for a pid" do
+      {:ok, pid} = GameServer.start_link(GameServer.via_tuple("hola"))
+      assert is_pid(pid)
+
+      assert "hola" == GameServer.get_uniq_name(pid)
+    end
+  end
+
+  describe "when everything fails" do
+    test "data is persisted in ETS" do
+      {:ok, pid} = GameServer.start_link()
+      game_name = GameServer.get_uniq_name(pid)
+      %Player{id: id1} = GameServer.join(pid)
+      %Player{id: id2} = GameServer.join(pid)
+
+      assert [{_name, current_state}] = :ets.lookup(:set_game_state, game_name)
+
+      assert length(current_state.players) == 2
+      assert Enum.map(current_state.players, fn %Player{id: id} -> id end) == [id2, id1]
+    end
+
+    test "if crashed, state is back there when restarted" do
```

Comment:
Here's the thing… if your `GameServer` crashes, it's probably BECAUSE the state is bad, and if you restart the GameServer process with bad state, it's probably going to crash again.

Also, boy it is confusing to use "set" as a noun everywhere. 😂

Reply:
What's the recommended path then? Check if the state is ok and, if so, restart it? Just crash it and leave the user with the game unended?